google-cloud-logging/lib/google/cloud/logging/async_writer.rb

```diff
@@ -149,23 +130,38 @@ def initialize logging, max_queue_size = DEFAULT_MAX_QUEUE_SIZE,
         #   async.write_entries entry
         #
         def write_entries entries, log_name: nil, resource: nil, labels: nil
-          ensure_thread
-          entries = Array(entries)
           synchronize do
-            raise "AsyncWriter has been stopped" unless writable?
-            queue_item = QueueItem.new entries, log_name, resource, labels
-            if @queue.empty? || !@queue.last.try_combine(queue_item)
-              @queue.push queue_item
-            end
-            @queue_size += entries.size
-            @queue_resource.broadcast
-            while @max_queue_size && @queue_size > @max_queue_size
-              @queue_resource.wait
+            raise "AsyncWriter has been stopped" if @stopped
+
+            @batch ||= Batch.new self
+            unless @batch.try_add(entries, log_name: log_name,
+                                           resource: resource, labels: labels)
+              publish_batch!
+              @batch = Batch.new self
+              @batch.add(entries, log_name: log_name, resource: resource,
+                                  labels: labels)
             end
+
+            init_resources!
+
+            publish_batch! if @batch.ready?
+
+            @cond.signal
```

Comment:
Are you sure about `signal` instead of `broadcast`? (If you're unsure, the correct answer is almost always `broadcast` to ensure that whatever thread needs to act will receive the signal. Signaling a single thread is generally for uncommon specialized cases, and even then is a lot harder to get right.)
Same question everywhere else `@cond.signal` is being called.

Reply:
There is only ever one thread, so my understanding is `signal` is appropriate. But I have no objection to changing this to `broadcast`.

Reply:
Updated to `broadcast`. PTAL and resolve this conversation if satisfied.